cardinal/world.go

```diff
@@ -0,0 +1,235 @@
+	go txh.Serve()
+	select {}
+}
+
+// RegisterSystems allows for the adding of multiple systems in a single call. See AddSystem for more details.
+// RegisterSystems adds the given system(s) to the world object so that the system will be executed at every
+// game tick. This Register method can be called multiple times.
+func (w *World) RegisterSystems(systems ...System) {
+	for _, system := range systems {
+		w.impl.AddSystem(func(world *ecs.World, queue *ecs.TransactionQueue) error {
+			return system(&World{impl: world}, &TransactionQueue{queue})
+		})
+	}
+}
+
+// RegisterComponents adds the given components to the game world. After components are added, entities
+// with these components may be created. This Register method must only be called once.
+func (w *World) RegisterComponents(components ...AnyComponentType) error {
+	return w.impl.RegisterComponents(toIComponentType(components)...)
+}
+
+// RegisterTransactions adds the given transactions to the game world. HTTP endpoints to queue up/execute these
+// transaction will automatically be created when StartGame is called. This Register method must only be called once.
+func (w *World) RegisterTransactions(txs ...AnyTransaction) error {
+	return w.impl.RegisterTransactions(toITransactionType(txs)...)
+}
+
+// RegisterReads adds the given read capabilities to the game world. HTTP endpoints to use these reads
+// will automatically be created when StartGame is called. This Register method must only be called once.
+func (w *World) RegisterReads(reads ...AnyReadType) error {
+	return w.impl.RegisterReads(toIReadType(reads)...)
+}
+
+// WorldOption represents an option that can be used to augment how the cardinal.World will be run.
+type WorldOption struct {
+	ecsOption      ecs.Option
+	serverOption   server.Option
+	cardinalOption func(*World)
+}
+
+// WithAdapter provides the world with communicate channels to the EVM base shard, enabling transaction storage and
+// transaction retrieval for state rebuilding purposes.
+func WithAdapter(adapter shard.Adapter) WorldOption {
+	return WorldOption{
+		ecsOption: ecs.WithAdapter(adapter),
+	}
+}
+
+// WithReceiptHistorySize specifies how many ticks worth of transaction receipts should be kept in memory. The default
+// is 10. A smaller number uses less memory, but limits the
+func WithReceiptHistorySize(size int) WorldOption {
+	return WorldOption{
+		ecsOption: ecs.WithReceiptHistorySize(size),
+	}
+}
+
+// WithNamespace sets the World's namespace. The default is "world". The namespace is used in the transaction
+// signing process.
+func WithNamespace(namespace string) WorldOption {
+	return WorldOption{
+		ecsOption: ecs.WithNamespace(namespace),
+	}
+}
+
+// WithHTTPServerPort specifies the port for the World's HTTP server. If omitted, the environment variable CARDINAL_PORT
+// will be used, and if that is unset, port 4040 will be used.
+func WithHTTPServerPort(port string) WorldOption {
```

Comment:
Thoughts on this being `WithPort` for consistency?

Reply:
I figured the longer name was more explicit in its behavior, but the function comment should be fine. done.